apps/mobile-wallet/src/features/send/modals/SelectAddressModal.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash } from '@alephium/shared'
+import { FlashList } from '@shopify/flash-list'
+
+import AddressBox from '~/components/AddressBox'
+import BottomModalFlashList from '~/features/modals/BottomModalFlashList'
+import { closeModal } from '~/features/modals/modalActions'
+import withModal from '~/features/modals/withModal'
+import { useAppDispatch, useAppSelector } from '~/hooks/redux'
+import { selectAllAddresses } from '~/store/addressesSlice'
+
+interface SelectAddressModalProps {
+  onAddressPress: (addressHash: AddressHash) => void
+}
+
+const SelectAddressModal = withModal<SelectAddressModalProps>(({ id, onAddressPress }) => {
+  const addresses = useAppSelector(selectAllAddresses)
+  const dispatch = useAppDispatch()
+
+  const handleAddressPress = (addressHash: AddressHash) => {
+    dispatch(closeModal({ id }))
+    onAddressPress(addressHash)
+  }
+
+  return (
+    <BottomModalFlashList modalId={id}>
+      {(props) => (
+        <FlashList
+          data={addresses}
+          keyExtractor={(item) => item.hash}
+          estimatedItemSize={70}
+          renderItem={({ item: address }) => (
+            <AddressBox
+              key={address.hash}
+              addressHash={address.hash}
+              style={{
+                marginBottom: 20
+              }}
+              onPress={() => handleAddressPress(address.hash)}
+            />
+          )}
```

Comment:
This is an example where we are using a render function, much more familiar to us than the new pattern you introduced

Reply:
I could go back to this one. I prefer the "new one" because I find it more readable (the DOM structure feels less deep and less cluttered to me). But yeah, I don't mind that much.

Reply:
This new pattern is very confusing to me and I would prefer reusing the `render` function prop convention that we are using everywhere else

Reply:
or we change it everywhere else and I learn to get used to a new pattern, I don't mind. I just don't like having 2 at the same time, I find it confusing.

Reply:
The main reason I don't like it is that we are kinda "overriding" the default from React, which is 
```ts
children?: ReactNode | undefined
```

A React developer is familiar with it. But an outsider of our repo will have to learn it and get used to it

I'd personally prefer to not touch the `children` prop. I wonder what your thoughts are. I find the benefits of readability (which I do not necessarily agree on since it took me by surprise when reviewing the PR) to be negligible compared to drawbacks.

Reply:
I prefer the first example above because we can clearly see the hierarchy (children being indented). It's harder to see in the second one. 

BUT I agree with your arguments and will update this in the follow up PR :)